pkg/cache/controller.go

```diff
@@ -67,5 +67,11 @@ func ClusterAwareKeyFunc(obj interface{}) (string, error) {
 
 // ToClusterAwareKey is a helper function that formats cluster, namespace, and name for key and index functions
 func ToClusterAwareKey(cluster, namespace, name string) string {
+	if cluster == "" {
```

Comment:
I'm not sure we want to change the key format based on presence/absence of values. Pros/cons? cc @stevekuznetsov

Reply:
I followed the similar approach as is used in upstream: 
https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/cache/internal/cache_reader.go#L213-L218

Reply:
We _can_ make this change, but I don't believe it's required. To keep this PR small, mind removing it, and just keeping the round_tripper.go change?

Reply:
a very similar change was introduced in the latest controller-runtime PR:
https://github.com/kcp-dev/controller-runtime/pull/13/files#diff-7e7ff90d22397cbc25370857a4c26baa1062866c5c9581e440d1e37603e45011R205-R214
Not sure why we need two ways/functions of doing the same thing.

Reply:
ok, now I see what the difference is - the one in controller-runtime is used in the generic cache_reader, but this function is used only in cluster-related functions (but at least it seems like so :thinking: )
Let me drop it then.

Reply:
dropped in [2d1c561](https://github.com/kcp-dev/apimachinery/pull/35/commits/2d1c56159a25a86ee6594bfd673aad48702099d2)